download/testharness.go

```diff
@@ -141,6 +141,99 @@ func tokenHandler(token string) http.HandlerFunc {
 	}
 }
 
+// withGitlabRegistryAuth sets up a token auth flow according to
+// the spec https://docs.docker.com/registry/spec/auth/token/.
+//
+// The flow is the same as for public registries but additionally
+// the request for fetching the token also has to be authenticated.
+//
+// The token issuing and validation differs between providers
+// and we only use a minimal version for testing.
+func withGitlabRegistryAuth() fixtureOpt {
```

Comment:
Authenticating against the token server is actually mentioned in an example in the [token authentication spec](https://docs.docker.com/registry/spec/auth/token/#example). It says:
> Whether the token server requires authentication is up to the policy of that access control provider. Some requests may require authentication to determine access [...] while others may not [...]

I think that even though Gitlab is an example that implements the token auth in this way, it is probably not the only provider doing it. From this perspective, a better name might be something like `withAuthenticatedTokenAuth` and we could mention gitlab in the comment?

Reply:
renamed it accordingly and extended the comment with gitlab as an example